lib/pbench/agent/results.py

```diff
@@ -14,45 +14,79 @@
 from pbench.common.utils import md5sum, validate_hostname
 
 
-class FileUploadError(Exception):
-    """Raised when the uploading of file to server has failed"""
-
-    pass
-
-
 class MakeResultTb:
     """MakeResultTb - Creates the result tar file.
 
-        TODO:  This is latent code -- it is currently unused and largely
-               untested, intended to support the future implementation of
-               a tool to replace pbench-make-result-tb.
     """
 
+    class AlreadyCopied(RuntimeError):
+        """Specific run time error raised when a result directory has a .copied
+        marker present.
+        """
+
+        pass
+
+    class BenchmarkRunning(RuntimeError):
+        """Specific run time error raised when a result directory has a .running
+        marker present.
+        """
+
+        pass
+
     def __init__(
         self,
         result_dir: str,
         target_dir: str,
+        controller: str,
         config: PbenchAgentConfig,
         logger: Logger,
     ):
-        """__init__ - Initializes the required attributes
+        """__init__ - Initializes the required attributes for making a tar ball
 
-            Args:
-                result_dir -- directory where tb file is collected.
-                target_dir -- directory where tar file needs to be moved.
-                config -- PbenchAgent config object
-                logger -- logger objects helps logging important details
+        Args:
+            result_dir -- directory where tb file is collected.
+            target_dir -- directory where tar file needs to be moved.
+            config -- PbenchAgent config object
+            logger -- logger objects helps logging important details
```

Comment:
Not sure why this is showing as a diff, since these lines didn't change; but if you pass this way again, you might fix the grammar of the "logger" description.